diff --git a/node_modules/react-native-tabbed-view-pager-android/android/src/main/java/is/uncommon/rn/widgets/ReactViewPager.java b/node_modules/react-native-tabbed-view-pager-android/android/src/main/java/is/uncommon/rn/widgets/ReactViewPager.java
index a018296..cb5262f 100644
--- a/node_modules/react-native-tabbed-view-pager-android/android/src/main/java/is/uncommon/rn/widgets/ReactViewPager.java
+++ b/node_modules/react-native-tabbed-view-pager-android/android/src/main/java/is/uncommon/rn/widgets/ReactViewPager.java
@@ -203,6 +203,25 @@ public class ReactViewPager extends ViewPager {
     mScrollEnabled = scrollEnabled;
   }
 
+  @Override
+  protected void onAttachedToWindow() {
+    super.onAttachedToWindow();
+    // The viewpager reset an internal flag on this method so we need to run another layout pass
+    // after attaching to window.
+    this.requestLayout();
+    post(measureAndLayout);
+  }
+
+  private final Runnable measureAndLayout = new Runnable() {
+    @Override
+    public void run() {
+      measure(
+              MeasureSpec.makeMeasureSpec(getWidth(), MeasureSpec.EXACTLY),
+              MeasureSpec.makeMeasureSpec(getHeight(), MeasureSpec.EXACTLY));
+      layout(getLeft(), getTop(), getRight(), getBottom());
+    }
+  };
+
   /*package*/ void addViewToAdapter(View child, int index) {
     getAdapter().addView(child, index);
   }
diff --git a/node_modules/react-native-tabbed-view-pager-android/android/src/main/java/is/uncommon/rn/widgets/TabbedViewPager.java b/node_modules/react-native-tabbed-view-pager-android/android/src/main/java/is/uncommon/rn/widgets/TabbedViewPager.java
index a71cd4d..e1486bc 100644
--- a/node_modules/react-native-tabbed-view-pager-android/android/src/main/java/is/uncommon/rn/widgets/TabbedViewPager.java
+++ b/node_modules/react-native-tabbed-view-pager-android/android/src/main/java/is/uncommon/rn/widgets/TabbedViewPager.java
@@ -86,6 +86,10 @@ public class TabbedViewPager extends LinearLayout {
     this.reactViewPager.removeViewFromAdapter(index);
   }
 
+  public void removeAllViewsFromAdapter() {
+    this.reactViewPager.removeAllViewsFromAdapter();
+  }
+
   public void setPageMargin(int i) {
     this.reactViewPager.setPageMargin(i);
   }
diff --git a/node_modules/react-native-tabbed-view-pager-android/android/src/main/java/is/uncommon/rn/widgets/TabbedViewPagerManager.java b/node_modules/react-native-tabbed-view-pager-android/android/src/main/java/is/uncommon/rn/widgets/TabbedViewPagerManager.java
index 2c04668..11098bd 100644
--- a/node_modules/react-native-tabbed-view-pager-android/android/src/main/java/is/uncommon/rn/widgets/TabbedViewPagerManager.java
+++ b/node_modules/react-native-tabbed-view-pager-android/android/src/main/java/is/uncommon/rn/widgets/TabbedViewPagerManager.java
@@ -149,6 +149,10 @@ import javax.annotation.Nullable;
     parent.removeViewFromAdapter(index);
   }
 
+  @Override public void removeAllViews(TabbedViewPager parent) {
+    parent.removeAllViewsFromAdapter();
+  }
+
   @ReactProp(name = "pageMargin", defaultFloat = 0)
   public void setPageMargin(TabbedViewPager pager, float margin) {
     pager.setPageMargin((int) PixelUtil.toPixelFromDIP(margin));
